src/wallet/interfaces.cpp

```diff
@@ -403,7 +404,24 @@ class WalletImpl : public Wallet
     CAmount getBalance() override { return GetBalance(*m_wallet).m_mine_trusted; }
     CAmount getAvailableBalance(const CCoinControl& coin_control) override
     {
-        return GetAvailableBalance(*m_wallet, &coin_control);
```

Comment:
> [8cd7730](https://github.com/bitcoin/bitcoin/commit/8cd77302305a2387368aabe8744b1859caf9956a): paging @ryanofsky: do you think we should move all this complexity here to the interface?

Right, probably it would be better to move the new code from the `wallet:WalletImpl::getAvailableBalance()` method in `src/wallet/interfaces.cpp` into the `wallet::GetAvailableBalance()` function in `src/wallet/spend.cpp` or into another function there like `wallet::GetSelectedCoinsBalance`. Reasoning is

- `src/wallet/interfaces.cpp` is supposed to define a public interface to wallet code that GUI code can call without accessing wallet internals. It is just supposed to forward calls from the public interface to private functions, and not really designed to contain coin selection code.
- It is easier to write unit tests for standalone `spend.cpp` functions than interface methods that require constructing a larger interface.
- It is easier to share code between RPC and GUI interfaces right now if code is standalone

Reply:
Sure. Having a proxy class is nice for any future IPC mechanism. All the protobuf/protocan/etc messages are simpler to craft in this way.

Based on how close we are to v25 branch-off, I'm thinking on whether should do it now or later :/. Otherwise I would push it right away.

Probably, could tackle it on a quick follow-up with https://github.com/bitcoin/bitcoin/pull/26699#discussion_r1156018577 and few other cleanups as well.